### DISCUSSION TITLE
Unable to get DateTime fields to save

### DISCUSSION THREAD

**Discussion by mindfulvector** [pocketbase/pocketbase#2142] [Q&A] [answered]
When I have a DateTime field set as required, values in the format I see given in other discussions are rejected and I can't get the record to save.

If I remove the required attribute from that field, it saves but without the value passed in for this field.

Here's an example curl request:

```
curl --location 'http://127.0.0.1:8090/api/collections/events/records' \
--header 'Authorization: Bearer xxx.xxx.xxx' \
--header 'Content-Type: application/json' \
--data '{"createdBy":"ooio9m3der6146l",
"scheduledAt":"20-03-2023 20:17:32.000Z",
"durationMinutes":"10",
"title":"zzzzz",
"description":"zzzzz"}]'
```

The saved record returned is:

```
{
    "collectionId": "2hgwaok8pebu9ql",
    "collectionName": "events",
    "created": "2023-03-24 03:17:19.168Z",
    "createdBy": "ooio9m3der6146l",
    "description": "zzzzz",
    "durationMinutes": 10,
    "id": "yjzp1o7olbw3hqt",
    "scheduledAt": "",
    "title": "zzzzz",
    "updated": "2023-03-24 03:17:19.168Z"
}
```

Any ideas why this isn't working?

Here are my migrations:
[pb_migrations.zip](https://github.com/pocketbase/pocketbase/files/11058152/pb_migrations.zip)

**Comment by ganigeorgiev** [comment, marked as answer]
In the curl example the datetime format must be `Y-m-d H:i:s.uZ`, eg. "2023-03-24 08:52:00.123Z".

_Note that system datetime fields, `created` and `updated`, cannot be changed by the users._

**Reply by mindfulvector** [reply]
Yep that was the problem, I knew I had to be doing something silly but just couldn't see that I had transposed the tokens for the date. >_< Thank you!

**Reply by nicetomytyuk** [reply]
@ganigeorgiev Hi there, I've noticed that even .getUpdated() or .getDateTime() when extending in JavaScript returns the same format, will there be any possibility to get the date in JS format `Y-m-dTH:i:s.uZ` or even the Date object directly in JavaScript?

**Reply by ganigeorgiev** [reply]
@nicetomytyuk The current `date` fields format cannot be changed. If you need a JS `Date` object, you can try with `new Date(yourDate.toString().replace(' ', 'T'))`.
